src/main/java/com/example/controller/SimulationController.java

```diff
@@ -195,6 +198,9 @@ protected void updateItem(AlgorithmType algorithmType, boolean empty) {
 
         pauseDisabledProperty.bind(Bindings.createBooleanBinding(() ->
                 !(!paused.get() && started.get() && !isFinished.get()), dependencies));
+
+        stopDisableProperty.bind(Bindings.createBooleanBinding(() ->
```

Comment:
Should stop button require paused state? Imo stop should be independent operation and should immediately break all ongoing actions like animations.

Reply:
I've tried that approach. However, it throws an InterruptedException that comes from another thread used by graphView update method.